src/main/java/com/github/teamfusion/spyglassplus/common/entity/SpyglassStandEntity.java

```diff
@@ -190,6 +221,24 @@ public ItemStack getSpyGlass() {
 		return this.entityData.get(DATA_SPYGLASS);
 	}
 
+	public void setOwner(Entity entity) {
+		if (entity == null) {
+			this.entityData.set(DATA_OWNER_ID, OptionalInt.empty());
+		} else {
+			this.entityData.set(DATA_OWNER_ID, OptionalInt.of(entity.getId()));
+		}
+	}
+
+	@Nullable
+	public Entity getOwner() {
+		OptionalInt optionalInt = this.entityData.get(DATA_OWNER_ID);
+		if (optionalInt.isPresent()) {
+			Entity entity = this.level.getEntity(optionalInt.getAsInt());
```

Comment:
If you change to UUID as mentioned above, you'll want to make sure to change this to getPlayerByUUID (the name of the method is at least similar to this)